app/services/arcgis_api/connection_factory.rb

```diff
@@ -0,0 +1,22 @@
+module ArcgisApi
+  # Stateless factory object that create a new Farady::COnnection object
+  class ConnectionFactory
```

Comment:
What is the purpose of this class? If there a reason we couldn't reuse `geocoder.faraday`?

Reply:
To easily inject a mock one that stub request on connection for testing purpose, also customize faraday middleware behaviours.

Reply:
@allthesignals , good point, this was to mock on faraday level, webmock covers much more.  It was adapted to the mock geocoder used in feature test. Let me c whether I can handle it better.